### DISCUSSION TITLE
SimilaritySearch Scores

### DISCUSSION THREAD

**Discussion by arkadyb** [tmc/langchaingo#791] [Q&A] [answered]
Would someone know, if there is a part of documentation that would be able to explain the `SimilaritySearch` results score values? I get plenty of results with different scores, but which one is better? How do i exactly treat the number? Thanks!

**Comment by CrazyWr** [comment, marked as answer]
You should learn some knowledge about VectorSearch

The score represents the similarity of the vectors. The similarity is measured by the score calculated by L2 (Euclidean distance), IP (inner product) or COSINE (cosine similarity). The output parameter score represents the similarity calculation score. 
Among them, the smaller the score calculated by Euclidean distance (L2), the more similar the search value is; while the larger the score calculated by cosine similarity (COSINE) and inner product (IP), the more similar the search value is.

**Comment by arkadyb** [comment]
Thanks for clarifying this @CrazyWr! I assume `lancgchaingo` in this case returns the L2 for the scrore?

**Comment by CrazyWr** [comment]
I'm not sure about this, I think it depends on which VectorStore you use, as far as I know Azure, pgvector, chroma use L2 by default

**Reply by arkadyb** [reply]
Ah, ok, i see. I use `pgvector` and `langchaingo` would seem to return the lowest value first.
